src/winter_onboarding_2021/fleet_management_service/handlers/cab.clj

```diff
@@ -0,0 +1,47 @@
+(ns winter-onboarding-2021.fleet-management-service.handlers.cab
+  (:require [ring.util.response :as response]
+            [clojure.spec.alpha :as s]
+            [winter-onboarding-2021.fleet-management-service.models.cab :as cab]
+            [winter-onboarding-2021.fleet-management-service.views.layout :as layout]
+            [winter-onboarding-2021.fleet-management-service.views.cab :as cab-views]))
+
+(s/def :cab/name string?)
+(s/def :cab/licence_plate string?)
+
+(s/def :cab/distance_travelled
+  (s/conformer
+   #(try (Long/parseLong %)
+         (catch Exception _ :clojure.spec.alpha/invalid))
+   str))
+
+(s/def ::create-cab-request
+  (s/keys :req-un [:cab/name
+                   :cab/licence_plate
+                   :cab/distance_travelled]))
+
+(def error-flash
+  {:flash {:error true
+           :style-class "alert alert-danger"
+           :message "Could not add cab, try again!"}})
+(def success-flash
+  {:flash {:success true
+           :style-class "alert alert-success"
+           :message "Cab added successfully!"}})
+
+(defn add [{:keys [multipart-params] :as request}]
+  (let [validated-cab (s/conform ::create-cab-request
```

Comment:
What do you think about moving the validation to a `pre` condition. If we use pre check here, we can add a middleware that returns 400 response on AssertionFailure exceptions. Are there other places where assertion failures can occur?

Reply:
Adding a `pre` condition to check for validation might prevent us from doing any side effects like redirecting to another page.
I think `pre` might be better for pure functions.